node/src/actors/rad_manager/handlers.rs

```diff
@@ -69,26 +72,27 @@ impl Handler<ResolveRA> for RadManager {
             }
         };
 
-        if let Some(timeout) = timeout {
-            // Add timeout, if there is one
-            // TODO: this timeout only works if there are no blocking operations.
-            // Since currently the execution of RADON is blocking this thread, we can only
-            // handle HTTP timeouts.
-            // A simple fix would be to offload computation to another thread, to avoid blocking
-            // the main thread. Then the timeout would apply to the message passing between threads.
-            Box::pin(
-                tokio::time::timeout(timeout, fut).map(|result| match result {
-                    Ok(Ok(x)) => Ok(x),
-                    Ok(Err(rad_error)) => Err(rad_error),
-                    Err(_elapsed) => Ok(RadonReport::from_result(
-                        Err(RadError::RetrieveTimeout),
-                        &ReportContext::default(),
-                    )),
-                }),
-            )
-        } else {
-            Box::pin(fut)
-        }
+        let timeout = match timeout {
+            None => MAX_RETRIEVAL_TIMEOUT,
+            Some(timeout_from_config) => std::cmp::min(timeout_from_config, MAX_RETRIEVAL_TIMEOUT),
+        };
+
+        // Add timeout, if there is one
```

Comment:
Now the timeout is not optional.

```suggestion
        // Add timeout
```

Reply:
In which cases was this timeout set to `None`?

Reply:
It is set to None when the miner set 0 as a timeout